### ISSUE TITLE
extract `ExtrinsicPayload` using api

### ISSUE THREAD

**Issue by nuel77** [polkadot-js/api#5398]
- [ ] Bug report
  - [ ] Feature request
  - [X] Support
  - [ ] Other

Using the API, how can I gracefully extract`ExtrinsicPayload` data for a given transaction? I have come across a few methods which include https://github.com/polkadot-js/api/issues/1421#issuecomment-540545590 and https://github.com/paritytech/txwrapper-core. Both seem kind of hacky, as I am not doing an "offline" transaction. just that I am signing the data using a different wallet (for example metamask) but will be online for the duration of create-sign-send. also, is there any way to send the transaction once it has been signed by the external wallet?  My end-goal here is to create an extrinsic then sign it using metamask and send to substrate chain.

**Comment by jacogr** [comment]
For any transaction, you can pass through a signer that is responsible for signing the actual payload. The api will call into that function with a constructed payload and expect a result back.

A simple example of a singer can be found here - https://github.com/polkadot-js/api/blob/master/packages/api/src/test/SingleAccountSigner.ts

It exposes the required interfaces, explicitly the `signPayload` is required to sign transactions.

Basically when sending, you would do -

```js
// our signer implementation that will be called into
import MySigner from './somewhere';

// some address
const ADDR = '1F...`;

...
// create a transaction, send it using the signer
await api.tx.balances.transfer('0x123...', 123).signAndSend(ADDR, { signer: new MySigner() });
```

**Comment by nuel77** [comment]
Thanks @jacogr so i just have to implement the a Signer class with signPayload fn for any wallets I am gonna use.
